neo/Network/P2P/RemoteNode.cs

```diff
@@ -146,10 +147,20 @@ protected override void OnReceive(object message)
             }
         }
 
+        public void Ping()
+        {
+            var ping = new PingPayload();
+            lastPingNonce = ping.Nonce;
+
+            SendMessage(Message.Create(MessageCommand.Ping, ping));
+        }
+
         private void OnPingPayload(PingPayload payload)
         {
-            if (payload.LastBlockIndex > LastBlockIndex)
+            if (payload.Nonce == lastPingNonce && payload.LastBlockIndex > LastBlockIndex)
```

Comment:
Why checking the `lastPingNonce`?

Reply:
What is the use of the nonce?

Reply:
`Nonce` is used to identify which ping the corresponding pong corresponds to.